addons/crm_livechat/models/chatbot_script_step.py

```diff
@@ -105,10 +105,12 @@ def _process_step_create_lead_and_forward(self, discuss_channel):
             and lead.filtered_domain(literal_eval(member.assignment_domain or "[]"))
         ]
         previous_operator = discuss_channel.livechat_operator_id
-        # sudo: im_livechat.channel - getting available operators is acceptable
-        users = discuss_channel.livechat_channel_id.sudo()._get_available_operators_by_livechat_channel(
-            self.env["res.users"].browse(assignable_user_ids)
-        )[discuss_channel.livechat_channel_id]
+        users = None
```

Comment:
As there's a check that users `is not None` later down the line to distinguish "restricted to no users" from "no restriction on users" (in _get_operator). Maybe have an empty recordset of users instead, as it seems we assume if there's a new operator it must be from a valid crm team etc..

Normally it shouldn't be an issue as we don't call _get_operator if there's no channel. But let's be on the safe side.